Documentation/ApiOverview/ContentElements/AddingYourOwnContentElements.rst

```diff
@@ -1,83 +1,79 @@
-Adding fields is done by :ref:`extending the TCA <extending-tca>`.
+Some data might need additional :ref:`ConfigureCE-DataProcessors`. Data
+processors are frequently used for example to process files
+(:ref:`t3tsref:FilesProcessor`) or to fetch related records
+(:ref:`t3tsref:DatabaseQueryProcessor`).
+
+A data processor can also be used to convert a string to an array,
+as is done for example in the *table* content element with the field `bodytext`.
 
-Depending on the data in the `tt_content` table,
-the data can be passed to the :ref:`cobj-fluidtemplate`.
-Some data might need additional processing by
-:ref:`data processor <t3tsref:cobj-fluidtemplate-properties-dataprocessing>` e.g. to resolve file relations.
-The content elements in the extension *fluid_styled_content* are using both methods.
-A data processor can be used to convert a string to an array,
-like done for the *table* content element with the field `bodytext`,
-or to fetch a related record, e.g. a FAL file.
 In these cases Fluid does not have to deal with these manipulations or transformation.
 
+You can find the example below in the TYPO3 Documentation Team extension
+`examples <https://extensions.typo3.org/extension/examples/>`__.
+
 
 Prerequisites
 =============
 
-Some of the following steps (specifically the ones using ``lib.contentElement``) require the system
-extension :ref:`fluid_styled_content <fsc:start>`. If you do not use **fluid_styled_content**, you
-must create and initialize the ``lib.contentElement`` TypoScript object yourself.
+The following examples require the system extension
+:ref:`fluid_styled_content <fsc:start>`.
+
+It can be via installed via composer with:
 
+.. code-block:: console
+
+   composer req typo3/cms-fluid-styled-content
+
+
+.. index:: Extension devolopment; Custom content element
 .. _AddingCE-use-an-extension:
 
-Use an Extension
+Use an extension
 ================
 
 We recommend to create your own extension for adding content elements.
-The following example uses the extension key `your_extension_key`.
+The following example uses the extension key `examples`.
 
-If you have plans to publish your extension, follow :ref:`extension-key-registration`.
+Here you can find information on how to
+:ref:`create an extension <extension-create-new>`.
 
-.. _AddingCE-PageTSconfig:
+
+.. index:: Content element; Registration
 .. _RegisterCE:
 .. _AddingCE-TCA-Overrides-tt_content:
 
-1. Register the Content Element
-===============================
+Register the content element
+============================
 
-First add the new content element to :guilabel:`New Content Element Wizard` and define its `CType` in PageTSconfig.
+First we need to define the key of the new content element type. We use
+`examples_newcontentelement` throughout the simple example.
 
-The example content element is called `yourextensionkey_newcontentelement`:
+First the key needs to be added to the select field `CType`. This will make it
```

Comment:
```suggestion
Next the key needs to be added to the select field `CType`. This will make it
```
Line 69 is a "first". Is this also a first? or the next step? I'm not sure.